src/libraries/System.Private.CoreLib/src/System/Reflection/InvokerEmitUtil.cs

```diff
@@ -0,0 +1,331 @@
+            Debug.Assert(parameters.Length <= MaxArgumentCount);
+
+            Type[] delegateParameters = new Type[3] { typeof(T), typeof(object), typeof(IntPtr*) };
+
+            // We could use the overload with 'owner' in order to associate to this module.
+            var dm = new DynamicMethod(
+                "InvokeStub_" + method.DeclaringType!.Name + "." + method.Name,
+                returnType: typeof(object),
+                delegateParameters,
+                owner: typeof(T),
+                skipVisibility: true);
+
+            ILGenerator il = dm.GetILGenerator();
+
+            if (parameters.Length == 0)
+            {
+                HandleThisPointer(il, method);
+                Invoke(il, method);
+            }
+            else
+            {
+                HandleThisPointer(il, method);
+                PushArguments(il, parameters, out NullableRefInfo[]? byRefNullables, out int byRefNullableCount);
+                Invoke(il, method);
+
+                if (byRefNullableCount > 0)
+                {
+                    Debug.Assert(byRefNullables != null); ;
+                    UpdateNullables(il, parameters, byRefNullables, byRefNullableCount);
+                }
+            }
+
+            HandleReturn(il, method);
+
+            return (InvokeFunc<T>)dm.CreateDelegate(typeof(InvokeFunc<T>));
+        }
+
+        private static void HandleThisPointer(ILGenerator il, MethodBase method)
+        {
+            bool emitNew = method is RuntimeConstructorInfo;
+            bool hasThis = !(emitNew || method.IsStatic);
+
+            if (hasThis)
+            {
+                il.Emit(OpCodes.Ldarg_1);
+                if (method.DeclaringType!.IsValueType)
+                {
+                    il.Emit(OpCodes.Unbox, method.DeclaringType);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Push each argument.
+        /// </summary>
+        private static void PushArguments(
+            ILGenerator il,
+            ParameterInfo[] parameters,
+            out NullableRefInfo[]? byRefNullables,
+            out int byRefNullablesCount
+            )
+        {
+            byRefNullables = null;
+            byRefNullablesCount = 0;
+
+            LocalBuilder? local_pArg = il.DeclareLocal(typeof(IntPtr*));
+            il.Emit(OpCodes.Ldarg_2);
+            il.Emit(OpCodes.Stloc_S, local_pArg);
+
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                RuntimeType parameterType = (RuntimeType)parameters[i].ParameterType;
+
+                // Get the argument as a ref
+                il.Emit(OpCodes.Ldloc_S, local_pArg);
+                il.Emit(OpCodes.Ldind_Ref);
+
+                if (parameterType.IsByRef)
+                {
+                    RuntimeType elementType = (RuntimeType)parameterType.GetElementType();
+
+                    if (elementType.IsNullableOfT)
```

Comment:
Both Nullable<T> and pointers should be handled by the type coercion in the static managed code. It should not be special-cased here.

Reply:
(Special-casing it here won't work for the eventual Span<TypedReference> overload.)

Reply:
Thanks. Coercing the nullable on the managed side will certainly remove the special case so I'll do that and also remove the special casing for that in the native code.